addons/hr_expense/models/hr_expense.py

```diff
@@ -1051,14 +1070,39 @@ def create(self, vals_list):
         })
         sheets = super(HrExpenseSheet, self.with_context(context)).create(vals_list)
         sheets.activity_update()
+        sheets._copy_expense_attachments()
         return sheets
 
+    def write(self, vals):
+        previous_lines = self.expense_line_ids
+        res = super().write(vals)
+
+        new_lines = self.expense_line_ids - previous_lines
+        if new_lines:
+            self._copy_expense_attachments(new_lines)
+
+        removed_line_ids = previous_lines - self.expense_line_ids
+        if removed_line_ids:
+            removed_line_ids._delete_expense_attachments()
+        return res
+
+
     @api.ondelete(at_uninstall=False)
     def _unlink_except_posted_or_paid(self):
         for expense in self:
             if expense.state in ['post', 'done']:
                 raise UserError(_('You cannot delete a posted or paid expense.'))
 
+    def _copy_expense_attachments(self, expense_ids=False):
+        for sheet in self:
+            expense_ids = expense_ids or sheet.expense_line_ids
+            attachment_ids = self.env['ir.attachment'].search([
+                ('res_model', '=', 'hr.expense'),
+                ('res_id', 'in', expense_ids.ids)
+            ])
+            for attachment in attachment_ids:
+                attachment.copy({'res_model': 'hr.expense.sheet', 'res_id': self.id, 'original_id': attachment.id})
```

Comment:
```suggestion
        expense_ids = expense_ids or self.expense_line_ids
        attachment_ids = self.env['ir.attachment'].search([
            ('res_model', '=', 'hr.expense'),
            ('res_id', 'in', expense_ids.ids)
        ])
        for attachment in attachment_ids:
            attachment.copy({'res_model': 'hr.expense.sheet', 'res_id': self.id, 'original_id': attachment.id})
```

However `'res_id': self.id` should be changed, in case there are multiple sheets